### PR TITLE
Adds information to configuration updates

### PR DESCRIPTION
**Pull request by dcalvoalonso**
When configuration updates are notified to IOTAs through `setConfigurationHandler` function, further information is needed to identify the configuration that is being modified:
- service
- subservice
- apikey
- resource

### CODE REVIEW

**Comment by fgalan** [comment]
Could you elaborate on the purpose of this PR, please? I mean, it is related with some existing issue? Does it have any impact in documentation? Should be covered by unit tests?

**Comment by dcalvoalonso** [comment]
This PR is related to the following issue in the LoRaWAN IOTA: https://github.com/Atos-Research-and-Innovation/IoTagent-LoRaWAN/issues/57.

The point is that in the case of this IOTA (and I understand that it may be also the situation for others), when a new configuration is provisioned or updated, we need to validate some special attributes and to create or modify the configuration of the agent to interact with some LoRaWAN servers.

Through https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/fiware-iotagent-lib.js#261, we can receive notifications for both situations. Nevertheless, the problem is that for updates only the [body of the update request](https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/fiware-iotagent-lib.js) is received and there it is not possible to identify at the IOTA which configuration / service is being modified.

Regarding your questions:
- Probably we need to update https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/usermanual.md#iotagentlibsetconfigurationhandler
- We could also add a couple of simple unit tests-

**Comment by fgalan** [comment]
So, I understand this is a kind of "extension" of the library to cope with requirements in LoRaWAN agent. No problem for that, as long as it doesn't have any impact in existing agentes (IOTA-UL, IOTA-JSON, IOTA-LWM2M and IOTA-Sigfox). Could you confirm that, pls?

With regards to the documentation and unit test you mention, that sounds good to me. Thanks!

**Comment by dcalvoalonso** [comment]
> So, I understand this is a kind of "extension" of the library to cope with requirements in LoRaWAN agent. No problem for that, as long as it doesn't have any impact in existing agentes (IOTA-UL, IOTA-JSON, IOTA-LWM2M and IOTA-Sigfox). Could you confirm that, pls?

I am almost sure (99.99%) that this does not have any impact in the rest of the IOTAs. Nevertheless, I will run some tests and come back with the results. :)

**Comment by dcalvoalonso** [comment]
> No problem for that, as long as it doesn't have any impact in existing agentes (IOTA-UL, IOTA-JSON, IOTA-LWM2M and IOTA-Sigfox). Could you confirm that, pls?

I have just run the tests for all the existing agents using this and everything is passed without any problem :)

**Comment by fgalan** [comment]
> I have just run the tests for all the existing agents using this and everything is passed without any problem :)

Great! :)

So I understand this PR is only pending of documentation and unit test changes (as commented at https://github.com/telefonicaid/iotagent-node-lib/pull/769#issuecomment-479927448), isn't it?

File: doc/usermanual.md [hunk 1]
```diff
@@ -382,6 +382,8 @@ callback with no parameters (this handler should only be used for reconfiguratio
 For the cases of multiple updates (a single Device Configuration POST that will create several device groups), the
 handler will be called once for each of the configurations (both in the case of the creations and the updates).
 
+The handler will be also called in the case of updates realted to configurations. In that situation, the `newConfiguration` parameter contains also the fields needed to identify the configuration to be updated, i.e., `service`, `subservice`, `resource` and `apikey`.
```
Comment (fgalan) [inline doc/usermanual.md]: Typo: realted

Comment (dcalvoalonso) [reply doc/usermanual.md]: Fixed with 24d48c072bd540adcec9a1e7ee2d0c32a1340b4a

**Comment by dcalvoalonso** [comment]
> So I understand this PR is only pending of documentation and unit test changes (as commented at #769 (comment)), isn't it?

Unit tests were updated with https://github.com/telefonicaid/iotagent-node-lib/pull/769/commits/0f70147a6e3c16d731a65bc7fdf2f615cfe874c5:

unit_tests

**Comment by fgalan** [comment]
> Unit tests were updated with 0f70147

What about documentation? Is it already updated?

(In addition, an entry in CHANGES_NEXT_RELEASE about the changes in this PR should be a good idea)

**Comment by dcalvoalonso** [comment]
> What about documentation? Is it already updated?

Documentation was updated with 24d48c072bd540adcec9a1e7ee2d0c32a1340b4a

**Comment by dcalvoalonso** [comment]
> (In addition, an entry in CHANGES_NEXT_RELEASE about the changes in this PR should be a good idea)

Fixed with 65fc59ad9d91cc19e5af14d122a2755c35e0fbc3

**Pull request merged**